sources/packages/web/src/router/InstitutionRoutes.ts

```diff
@@ -1,25 +1,68 @@
 import { RouteRecordRaw } from "vue-router";
 import InstitutionDashboard from "../views/institution/InstitutionDashboard.vue";
 import AppInstitution from "../views/institution/AppInstitution.vue";
-import { InstitutionRoutesConst } from "../constants/routes/RouteConstants";
+import {
+  InstitutionRoutesConst,
+  SharedRouteConst,
+} from "../constants/routes/RouteConstants";
 import Login from "../views/institution/Login.vue";
+import WorkInProgress from "../views/WorkInProgress.vue";
+import { AppConfigService } from "../services/AppConfigService";
+import { ClientIdType } from "../types/contracts/ConfigContract";
+import { AuthStatus, AppRoutes } from "../types";
 
 export const institutionRoutes: Array<RouteRecordRaw> = [
   {
-    path: "/institution",
+    path: AppRoutes.InstitutionRoot,
     name: InstitutionRoutesConst.APP_INSTITUTION,
     component: AppInstitution,
     children: [
       {
-        path: "login",
+        path: AppRoutes.Login,
         name: InstitutionRoutesConst.LOGIN,
         component: Login,
       },
       {
-        path: "institution-dashboard",
+        path: AppRoutes.InstitutionDashboard,
         name: InstitutionRoutesConst.INSTITUTION_DASHBOARD,
         component: InstitutionDashboard,
       },
+      {
+        path: AppRoutes.InstitutionProfile,
+        name: InstitutionRoutesConst.INSTITUTION_PROFILE,
+        component: WorkInProgress,
+      },
     ],
+    beforeEnter: (to, from, next) => {
```

Comment:
I would consider two things for the "beforeEnter" method.
1 - Move the method definition outside the array for better readability.
2 - Since we have a RouterHelper, maybe this logic could be moved there.

Reply:
I will suggest Route Helper to handle this but as refactor of code with better access management.